libethereum/BlockQueue.h

```diff
@@ -67,6 +67,117 @@ enum class QueueStatus
 
 std::ostream& operator<< (std::ostream& os, QueueStatus const& obj);
 
+template<class T>
+class SizedBlockQueue
+{
+public:
+	std::size_t count() const
+	{
+		return m_queue.size();
+	}
+
+	std::size_t size() const
+	{
+		return m_size;
+	}
+	
+	bool isEmpty() const
+	{
+		return m_queue.empty();
+	}
+
+	h256 nextHash() const
+	{
+		return m_queue.front().verified.info.sha3Uncles();
+	}
+
+	T const& next() const
+	{
+		return m_queue.front();
+	}
+
+	template<class Op>
+	void forEach(Op _op) const
```

Comment:
Would it be better to implement a boost range compatible interface so that `boost::for_each(queue, []{..})` can be used as easily as any other boost range algorithms?